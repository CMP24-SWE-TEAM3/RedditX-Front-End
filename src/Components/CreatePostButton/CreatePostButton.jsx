// imports
import React from 'react';
import {HiOutlinePlus} from 'react-icons/hi';
import {GlobalButtonStyled} from "Components/GlobalButton/GlobalButton.styled";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";

/**
 * Component that displays the profile button
 *
 * @param props to test against properties
 *
 * @returns {React.Component}
 */
const CreatePostButton = (props) => {
    return (
        <OverlayTrigger
            key={'bottom'}
            placement={'bottom'}
            overlay={
                <Tooltip id={`tooltip-bottom`}>Create Post</Tooltip>}>
            <GlobalButtonStyled onClick={props.onshow}>
                <HiOutlinePlus/>
            </GlobalButtonStyled>
        </OverlayTrigger>
    );
};

export default CreatePostButton;
