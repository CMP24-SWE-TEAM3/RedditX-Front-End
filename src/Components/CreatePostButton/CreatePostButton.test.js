import {shallow} from 'enzyme';

// import Components
import CreatePostButton from "./CreatePostButton";

describe("Create Post button Component", () => {
    it("this is a test for Create post button", () => {
        expect(shallow(<CreatePostButton/>)).toMatchSnapshot();

    });
});