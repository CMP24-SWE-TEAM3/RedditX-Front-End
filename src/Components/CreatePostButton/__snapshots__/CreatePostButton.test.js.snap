// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Post button Component this is a test for Create post button 1`] = `ShallowWrapper {}`;
